mk/cache-test.mk

```diff
@@ -0,0 +1,42 @@
+CACHE_TEST_DIR := tests/cache
+CACHE_BUILD_DIR := build/cache
+TARGET := test_cache.out
+SOURCE := src
+
+CACHE_OBJS := cache.o \
+        test_cache.o
+
+CACHE_OBJS := $(addprefix $(CACHE_BUILD_DIR)/, $(CACHE_OBJS))
+OBJS += $(CACHE_OBJS)
+deps += $(CACHE_OBJS:%.o=%.o.d)
+
+CACHE_CHECK_ELF_FILES := \
+	cache_new \
+	cache_put \
+	cache_get \
+	cache_lru_replace \
+	cache_lfu_replace \
+	cache_lfu_ghost_replace \
+
+cache :$(TARGET)
+	$(Q)$(foreach e,$(CACHE_CHECK_ELF_FILES),\
+	    $(PRINTF) "Running $(e) ... "; \
+	    if [ "$(strip $(shell cat $(CACHE_TEST_DIR)/$(e).expect))" = "$(strip $(shell ./$(CACHE_BUILD_DIR)/$(TARGET) $(CACHE_TEST_DIR)/$(e).in))" ]; then \
```

Comment:
Shouldn't you compare `.out` (the output) and `.expect` (the expected result)?

Reply:
I campare the output and .expect directly.

Reply:
How about using [cmp(1)](https://man7.org/linux/man-pages/man1/cmp.1.html)?